test/buffers_utest.cpp

```diff
@@ -63,222 +55,156 @@ class BuffersUtest {
             }
             vec->set(i, buf);
         }
-        // vec.dump();
 
         return vec;
     }
+};
 
-    void buffers_utest1()
-    {
-        std::cout << "buffers_utest1\n";
-
-        int n = 16;
-        int begin = 5;
-        int end = 12;
-        int size = 32;
-        int i, j;
-
-        quadiron::vec::Buffers<T>* vec1 = gen_buffers_rand_data(n, size);
-        std::vector<T*>* mem1 = vec1->get_mem();
-        // vec1->dump();
-
-        quadiron::vec::Buffers<T> vec2(vec1, begin, end);
-        std::vector<T*>* mem2 = vec2.get_mem();
-        assert(vec2.get_n() == end - begin);
-        assert(vec2.get_size() == vec1->get_size());
-        for (i = 0; i < end - begin; i++) {
-            for (j = 0; j < size; j++) {
-                mem2->at(i)[j] = mem1->at(i + begin)[j];
-            }
-        }
-        // vec2.dump();
+using TestedTypes = ::testing::Types<uint32_t, uint64_t>;
+TYPED_TEST_CASE(BuffersTest, TestedTypes);
 
-        std::vector<T*> mem3(end - begin, nullptr);
-        for (int i = 0; i < end - begin; i++) {
-            mem3[i] = mem1->at(i + begin);
-        }
-        quadiron::vec::Buffers<T> vec3(end - begin, size, &mem3);
-        // vec3.dump();
+TYPED_TEST(BuffersTest, Test1) // NOLINT
```

Comment:
@lamphamsy If you have a better name than Test1, I'll be glad to rename it (it was named utest1 before so I kept it like that)

Reply:
It could be renamed as: `Check constructors`

Reply:
Done